### PR TITLE
add file list func

### PR DESCRIPTION
**Pull request by ehanson8**
#### What does this PR do?
Adding functions to build a file list from a local file system as well as a web directory. These functions are the first components of a workflow to ingest new content into DSpace. I haven't figured how I want to test the local file system function yet but there is a test for the web directory function.

#### Includes new or updated dependencies?
YES

### CODE REVIEW

**Comment by ehanson8** [comment]
Added a test for the local file list function

File: dsaps/models.py [hunk 1]
```diff
@@ -123,6 +125,28 @@ class MetadataEntry(BaseRecord):
     language = Field()
 
 
+def build_file_list_local(directory, file_extension):
+    """Build list of files in local directory."""
+    file_list = {}
+    for root, dirs, files in os.walk(directory, topdown=True):
+        for file in files:
+            if file.endswith(file_extension):
+                full_file_path = os.path.join(root, file).replace('\\', '/')
```
Comment (gravesm) [inline dsaps/models.py]: What's the reason for using the replace here? I'm assuming it has something to do with Windows, but I'd like to better understand the requirement this is solving.

Comment (ehanson8) [reply dsaps/models.py]: I think this is legacy code from my last job that I copied without thinking, it doesn't seem necessary so I'll remove it

Comment (gravesm) [reply dsaps/models.py]: In this case, I would just get rid of this function entirely. The standard library already provides the ability to recursively generate a list of files matching a specific extension:

```python
import glob

files = glob.glob('/path/to/search/**/*.pdf', recursive=True)
```

If there are potentially going to be lots (thousands) of files I'd suggest using `iglob` instead. This functionality is also in the [pathlib](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) module.

File: tests/test_models.py [hunk 2]
```diff
@@ -68,7 +71,36 @@ def test__pop_inst(client):
...
 
 def test__build_uuid_list(client):
+    """Test _build_uuid_list function."""
     rec_obj = {'items': [{'uuid': '1234'}]}
     children = 'items'
     child_list = client._build_uuid_list(rec_obj, children)
     assert '1234' in child_list
+
+
+def test_build_file_list_local():
```
Comment (gravesm) [inline tests/test_models.py]: If you remove the function that this is testing this test would obviously no longer be needed. For future reference though, I'd suggest using one of pytest's [temp dir fixtures](https://docs.pytest.org/en/latest/tmpdir.html). I've found a few isolated cases where they don't quite fit, but they will generally be a better option than trying to manage temp files and temp dirs on your own.

File: dsaps/models.py (hunk 1 above)
Comment (ehanson8) [reply dsaps/models.py]: Makes sense, I will save this for when I create a click command for the full workflow. Thanks!

File: tests/test_models.py (hunk 2 above)
Comment (ehanson8) [reply tests/test_models.py]: Great, though I hope I don't have consider too many local file operations in the future

**Pull request merged**